### PR TITLE
For jib mode use registry properties for build "from" image section

### LINKED ISSUE #2077: Jib mode ignore registry settings for "from" image name section while building
### Component

JKube Kit

### Is your enhancement related to a problem? Please describe

Exec command: 
> mvn k8s:build "-Djkube.build.strategy=jib"

or

> mvn k8s:build "-Djkube.build.strategy=jib" "-Djkube.docker.registry=myRegistry"

Settings:
```xml
<configuration>
   <registry>myRegistry</registry>
   <images>
      <image>
         <name>imageName</name>
         <registry>myRegistry</registry>
         <build>
            <from>baseImageFromMyRegistry:tag</from>
         </build>
          ...
      </image>
   </images>
</configuration>
```

Current Behave:
Try to pull baseImageFromMyRegistry from default docker registry.
Expected Behave:
Try to pull baseImageFromMyRegistry from myRegistry registry

### Describe the solution you'd like

Make the same behave of getting registry information as
> mvn k8s:build

Like: https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin/#registry for example.

### Describe alternatives you've considered

Set full image name to "from" section is workaround.

```xml
<configuration>
   <images>
      <image>
         <name>imageName</name>
         <build>
            <from>myRegistry/baseImageFromMyRegistry:tag</from>
         </build>
          ...
      </image>
   </images>
</configuration>
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Osmyslitelny**
## Description

Fixes #2077

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by rohanKanojia** [comment]
Welcome, Thanks a lot for contributing to Eclipse JKube :+1:,

In order to contribute to Eclipse, you would need to create an [Eclipse Account](https://accounts.eclipse.org/) and Sign [Eclipse Contributor Agreement](https://accounts.eclipse.org/user/login?destination=user/eca). Once you've created an eclipse account you would need to update your commit message with a `Signed-off-by:` footer. You can do it like this:
```
# (Optional): Configure Git User and Email
git config user.name "Your Name"
git config user.email "your_email@example.com"

# Amend your previous commit with signed off footer
git commit --amend --signoff
```

**Comment by rohanKanojia** [comment]
PR looks good, Do you think we should also handle `jkube.docker.registry` property or this will suffice?

**Comment by Osmyslitelny** [comment]
> PR looks good, Do you think we should also handle `jkube.docker.registry` property or this will suffice?

I'm not sure because I haven't check it, but I thought `imageConfig.getRegistry()` based on parsed `jkube.docker.registry`.</br>
Also I didn't get why this class uses `jkube.docker.push.registry` (not `jkube.docker.pull.registry` or `jkube.docker.registry`).</br> I've decided to use simplest and lowest code variant of implementation that I saw.
Maybe it will be better to make more correct parser logic, but current variant should work.

**Comment by rohanKanojia** [comment]
@Osmyslitelny : ECA check seems to be failing. Did you follow the steps I mentioned in https://github.com/eclipse/jkube/pull/2082#issuecomment-1469370905 to complete the legal requirements for contributing to Eclipse Foundation project?

**Comment by Osmyslitelny** [comment]
> @Osmyslitelny : ECA check seems to be failing. Did you follow the steps I mentioned in [#2082 (comment)](https://github.com/eclipse/jkube/pull/2082#issuecomment-1469370905) to complete the legal requirements for contributing to Eclipse Foundation project?

Actually, I don't have any idea what I'm doing wrong.  Because for me my last commit text looks correctly. https://github.com/eclipse/jkube/commit/1abd75533e6778bf134d2bb37a0ea6be6e173d0a

**Comment by rohanKanojia** [comment]
Did you use that email to create an [Eclipse Account](https://accounts.eclipse.org/) and Sign [Eclipse Contributor Agreement](https://accounts.eclipse.org/user/login?destination=user/eca).?

**Comment by Osmyslitelny** [comment]
> Did you use that email to create an [Eclipse Account](https://accounts.eclipse.org/) and Sign [Eclipse Contributor Agreement](https://accounts.eclipse.org/user/login?destination=user/eca)?

I did a lot attemptions and got fail result. Now I've done the same actions and *magic* - step is green. Don't know what was that but resolved.

**Comment by rohanKanojia** [comment]
Thanks a lot.

File: jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildService.java [hunk 1]
```diff
@@ -134,9 +136,22 @@ static ImageConfiguration prependRegistry(ImageConfiguration imageConfiguration,
...
         }
+        prependRegistryForFromImage(imageConfiguration, registry);
         return imageConfiguration;
     }
 
+    private static ImageConfiguration prependRegistryForFromImage(ImageConfiguration imageConfig, String registry) {
+        BuildConfiguration buildConfiguration = imageConfig.getBuildConfiguration();
+        String fromImage = BuildUtil.extractBaseFromConfiguration(buildConfiguration);
+        String imageRegistry = fromImage != null ? new ImageName(fromImage).getRegistry() : null;
+        final String actualRegistry = EnvUtil.firstRegistryOf(imageRegistry, registry, imageConfig.getRegistry());
```
Comment (sunix) [inline jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildService.java:147]: Maybe not execute this if imageRegistry not null?

Comment (Osmyslitelny) [reply jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildService.java:147]: m? Why? We get list of potential registry sources. Then we get 1st from the query that is valid. As common logic for this plugin.

Comment (sunix) [reply jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildService.java:147]: `actualRegistry` is used in the next line if `imageRegistry` is null. So if `imageRegistry` is not null, evaluating `actualRegistry` is useless as it won't be used afterward.

Comment (Osmyslitelny) [reply jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildService.java:147]: You are absolutely right, thank you.
Fixed (removed unnecessary condition).

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**